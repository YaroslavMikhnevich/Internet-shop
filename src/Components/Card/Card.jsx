import React, { useState } from 'react';
import ReactDOM from 'react-dom/client';
import './Card.scss'
import { RiShoppingBagLine } from 'react-icons/ri'
import CardList from '../CardList/CardList';
import QuantityCard from '../QuantityCard/QuantityCard';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

export default function Card(props) {
    const [activeCard, setactiveCard] = useState(false)

    const orders = useSelector(state => state.items.orders);

    return (
        <div className='card-block'>
            <button onClick={() => { setactiveCard(!activeCard) }}>
                {!activeCard ? <RiShoppingBagLine
                    className={`shopping-bag ${activeCard && 'active'}`} /> : ''}

                {!activeCard && <QuantityCard
                    quantity={orders.length}
                />}

                {activeCard && <CardList
                    state={activeCard}
                    funcs={[setactiveCard]}
                    message={'Упс! Кошик порожній'}
                    orders={orders} />}
            </button>
        </div>
    )
}

Card.propTypes = {
    delateOrder: PropTypes.func,
    orders: PropTypes.array,
        }

