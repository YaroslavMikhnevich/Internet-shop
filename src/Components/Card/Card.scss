.card-block {
    display: flex;
    align-items: center;

  
    button {
        position: relative;
        cursor: pointer;
        border: 0;
        background-color: #45bcec;
        text-align: center;



        .shopping-bag{
            color: white;
            font-size: 45px;
        }

        .shopping-bag:hover{
            color: rgb(220, 22, 22);
            
        }

        .active{
            color: rgb(220, 22, 22);
        }

        .close-order-window  {
            position: absolute;
            top: -22px;
            right: -20px;
            border-radius: 50%;
            background-color: rgba(0, 0, 0, 0.3803921569);
            padding: 7px;
            font-size: 32px;

            &:hover{
                background-color: rgba(0, 0, 0, 0.599);

            }
        }


    }





    .quantity-items {
        display: flex;
        position: absolute;
        padding: 3px;
        justify-content: center;
        bottom: 0px;
        right: 3px;
        width: 15px;
        height: 15px;
        border-radius: 50%;
        background-color: red;
        color: white;

    }

}