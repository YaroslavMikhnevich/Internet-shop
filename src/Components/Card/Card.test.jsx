import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import Card from './Card';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Card component', () => {
  it('should render correctly', () => {
    useSelector.mockImplementation(() => ({
      orders: [{ id: 1, name: 'Product 1', price: 10 }],
    }));

    const component = renderer.create(<Card />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});