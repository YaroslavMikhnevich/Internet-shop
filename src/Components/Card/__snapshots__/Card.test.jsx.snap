// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
<div
  className="card-block"
>
  <button
    onClick={[Function]}
  >
    <svg
      className="shopping-bag false"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M7 8V6a5 5 0 1 1 10 0v2h3a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V9a1 1 0 0 1 1-1h3zm0 2H5v10h14V10h-2v2h-2v-2H9v2H7v-2zm2-2h6V6a3 3 0 0 0-6 0v2z"
        />
      </g>
    </svg>
    <p
      className="quantity-card"
    />
  </button>
</div>
`;
